conans/client/build/cmake.py

```diff
@@ -215,13 +215,17 @@ def _run(self, command):
         the_os = self._settings.get_safe("os")
         is_clangcl = the_os == "Windows" and compiler == "clang"
         is_msvc = compiler == "Visual Studio"
-        if ((is_msvc or is_clangcl) and platform.system() == "Windows" and
-                self.generator in ["Ninja", "NMake Makefiles", "NMake Makefiles JOM"]):
-            vcvars_dict = tools.vcvars_dict(self._settings, force=True, filter_known_paths=False,
-                                            output=self._conanfile.output)
-            with _environment_add(vcvars_dict, post=self._append_vcvars):
-                self._conanfile.run(command)
-        else:
+        is_intel = compiler == "intel"
+        context = tools.no_op()
+        if self.generator in ["Ninja", "NMake Makefiles", "NMake Makefiles JOM", "Unix Makefiles"]:
```

Comment:
The introduction of new ``Unix Makefiles`` for users that are using pure ``clang`` on Windows might be breaking, as it will suddenly inject the ``vcvars`` environment, something that it was not doing before.

Reply:
yeah, I've taken it from the original PR (https://github.com/ohanar/conan/commit/7a62f3ac35573fdc6aa6a25d44111b87c726cb90)
probably, need to change it a bit

Reply:
changed to keep the previous behavior